reactioncommerce/reaction
# feat: GQL tax rates

Resolves #5761  
Impact: **major**  
Type: **feature**

## Issue

No GQL queries and mutation for tax rates.

## Solution

Add GQL queries and mutations for tax rates.

## Breaking changes

## Testing
1. Add a tax rate
1. Update a tax rate
1. Query for tax rates
1. Delete a tax rate
1. Query again for tax rates and ensure the one you added is gone


### Here are the mutations/queries, you'll need to update the IDs to your shop and tax rate documents.

```graphql
query Queries {
  primaryShopId
  taxRates(shopId: "cmVhY3Rpb24vc2hvcDpDN2NNS3ptOXVUVGtHbnNxcw==") {
		nodes {
      _id
    	rate
      shop {
        _id
      }
      postal
      country
      taxCode
      postal
    }
  }
}

mutation AddTaxRate {
  createTaxRate(input: {
    region: "CA",
    rate: 0.10,
    shopId: "cmVhY3Rpb24vc2hvcDpDN2NNS3ptOXVUVGtHbnNxcw==",
    country: "USA",
    postal: "90066",
    taxCode: "what?"
  }) {
    taxRate {
      _id
      rate
      shop {
        _id
      }
      postal
      country
      taxCode
      postal
    }
  }
}

mutation UpdateTaxRate {
  updateTaxRate(input: {
    taxRateId: "cmVhY3Rpb24vdGF4UmF0ZTpvcEY4VGg5ZDJUS0Rqcm5yOQ==",
    shopId: "cmVhY3Rpb24vc2hvcDpDN2NNS3ptOXVUVGtHbnNxcw==",
    region: "CA",
    rate: 0.40,
    country: "USA",
    postal: "90066",
    taxCode: "what?"
  }) {
    taxRate {
      _id
      rate
      shop {
        _id
      }
      postal
      country
      taxCode
      postal
    }
  }
}

mutation DeleteTaxRate {
  deleteTaxRate(input: {
    taxRateId: "cmVhY3Rpb24vdGF4UmF0ZTo0YWc1OUdNOXl0WGNnTGlwRw==",
    shopId: "cmVhY3Rpb24vc2hvcDpDN2NNS3ptOXVUVGtHbnNxcw=="
  }) {
    taxRate {
      _id
      rate
      shop {
        _id
      }
      postal
      country
      taxCode
      postal
    }
  }
}
```
